### DISCUSSION TITLE
Getting Tesla Battery Modules connectet to Victron without changing BMB

### DISCUSSION THREAD

**Discussion by GaryCooper23** [Louisvdw/dbus-serialbattery#772] [Ideas]
Iam seeing huge potential to get this BMS [https://github.com/collin80/TeslaBMS](https://github.com/collin80/TeslaBMS) integrated with Venus OS trough dbus-serialbattery.
Currently iam working on changing the balancing from the default code to another because the art of balancing is not the best. While ChatGPT is not great i want to spread the Idea, maybe someone can help:
My idea is to change the TeslaBMS code in the serial output an make a battery_template for the BMS. The only Hardware you need is then an Arduino Due which is about 47€ and some cables. This will replace the expensive 3rd Party BMS Solutions and will be connected to Venus OS.

**Comment by GaryCooper23** [comment]
Arduino Due and only 4 Wires to the BMB. I connected it to the standard Molex connectors:
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/115779675/8b60d98b-71cc-4c9a-9364-47bf79dea8ff)

Detailed Serial Output of Cells:
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/115779675/c9f6f9c3-97a8-425c-a96a-cc4e81926658)

Current Serial Options:
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/115779675/e823c7d1-cd4c-4c59-9603-7380aa07f365)

My Tesla Battery Modules:
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/115779675/8091bd27-d076-4e4f-94f0-287ff1a5da4f)

Molex Connectors which I wired together and connected TxRx to Arduino Due:
![image](https://github.com/Louisvdw/dbus-serialbattery/assets/115779675/b31071d2-861b-4aa9-96f9-7b724a3479ed)

**Reply by samehdababneh** [reply]
Hello, any update on the above project?
